### DISCUSSION TITLE
Way to pass dynamic data for validation only?

### DISCUSSION THREAD

**Discussion by rockisch** [pydantic/pydantic#5701] [Question] [answered]
Hi! I need a pydantic model to validate some data based on an external variable that has nothing to do with the data itself. Is there any way to pass dynamic values to a validator without subclassing the main class and defining a method on the subclass at runtime? Maybe a way to set a dynamic config object?

Here's an example with a hypothetical API that would solve my issue where you can dynamically set an `extra` dict to a validation call:

```py
class TestModel(BaseModel):
    value: str

    @validator('value')
    def check_str(cls, v, extra):
        if v not in extra['choices']:
            raise ValueError
        return v

# Dynamically set 'choices'
if cond:
    choices = ['a', 'b']
else:
    choices = ['c', 'd']

TestModel.with_extra(choices=choices)(value='a')
```

Is there any way to do something like this?

**Comment by dmontagu** [comment, marked as answer]
In v2 we have  validation context, which will function like extra. Look at the `context` argument to `TestModel.model_validate`. Prior to v2, I would recommend creating a function that retrieves the dynamic context, and using that inside your validator. You may be able to achieve an API somewhat like your `.with_extra` through the use of a ContextVar.

**Reply by rockisch** [reply]
Ah, knowing v2 will add support for this is more than enough for me.

I'm doing dynamic subclassing at the moment, and that works fine, it's just a bit more verbose.

**Reply by YanSte** [reply]
@rockisch Could you give you an example, I'm interested?
